modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +41,618 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
+
+[source,xml, linenums]
+----
+<root>
+
+  <h:table xmlns:h="http://www.w3.org/TR/html4/">
+    <h:tr>
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </h:table>
+
+  <f:table xmlns:f="https://www.w3schools.com/furniture">
+    <f:tr>
+      <f:name>African Coffee Table</f:name>
+      <f:width>80</f:width>
+      <f:length>120</f:length>
+    </f:tr>
+  </f:table>
+
+</root>
+----
+
+[[namespace_variable]]
+==== Source
+
+The DataWeave script selects XML content from specified namespaces. The
+script's header creates namespace (`ns`) variables `html` and `furniture`
+to store namespaces from the input payload. To select the children of `h:table`
+into element `a` and the children of `f:table` elements into element `b`, the
+script uses the `html` and `furniture` namespace variables with a
+namespace selector (`#`) and local part (`table`).
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+ns html http://www.w3.org/TR/html4/
+ns furniture https://www.w3schools.com/furniture
+---
+root: {
+     a: payload.root.html#table,
+     b: payload.root.furniture#table
+ }
+----
+
+==== Output
+
+The script outputs children of the `h:table` element under element `a` and
+the children of the `f:table` under element `b`.
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+  <b>
+    <f:tr xmlns:f="https://www.w3schools.com/furniture">
+      <f:name>African Coffee Table</f:name>
+      <f:width>80</f:width>
+      <f:length>120</f:length>
+    </f:tr>
+  </b>
+</root>
+----
+
+=== Example
+
+This example selects XML values using a local name `table` from the `h:table`
+element.
+
+==== Input
+
+The input payload contains XML with a `table` element that contains
+the namespace `"http://www.w3.org/TR/html4/"`.
+
+[source,xml, linenums]
+----
+<root>
+  <h:table xmlns:h="http://www.w3.org/TR/html4/">
+    <h:tr>
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </h:table>
+</root>
+----
+
+==== Source
+
+The DataWeave script selects values of children in the `table` element
+by using a local name of the XML element `table` instead of its namespace.
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+---
+root: { a: payload.root.table }
+----
+
+==== Output
+
+The script outputs XML that contains the values of children in the
+`table` element.
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+</root>
+----
+
+[[attribute_selector]]
+== Attribute Selector
+
+The attribute selector returns the first attribute value that matches the
+selected name expression. If no key matches, the selection returns the value
+`null`.
+
```

Comment:
Missing ===Example heading...although I kind of like it this way instead of the way we have been doing it, because we omit that level of heading and can promote the subs up a level....